rosbag2_py/src/rosbag2_py/_transport.cpp

```diff
@@ -127,14 +128,43 @@ typedef OptionsWrapper<rosbag2_transport::RecordOptions> RecordOptions;
 namespace rosbag2_py
 {
 
+
+class Arguments {
+public:
+  Arguments(const std::vector<std::string>& args) : arguments_(args) {
+    std::for_each(arguments_.begin(), arguments_.end(),
+        [this](const std::string& arg){
+          pointers_.push_back(const_cast<char*>(arg.c_str()));
+        }
+    );
+    pointers_.push_back(NULL);
+  }
+
+  char ** argv()
+  {
+    return arguments_.empty() ? nullptr : pointers_.data();
+  }
+
+  int argc() const
+  {
+    return arguments_.size();
+  }
+
+private:
+  std::vector<std::string> arguments_;
+  std::vector<char*> pointers_;
+};
+
+
 class Player
 {
 public:
   using SignalHandlerType = void (*)(int);
 
-  Player()
+  Player(const std::string& log_level)
```

Comment:
```suggestion
explicit Player(const std::string & log_level)
```
Use `explicit` qualifier to avoid occasional type conversion.

Reply:
done